src/NHibernate/Engine/ForeignKeys.cs

```diff
@@ -179,9 +179,17 @@ public static bool IsNotTransientSlow(string entityName, object entity, ISession
 			}
 
 			// let the interceptor inspect the instance to decide
-			// let the persister inspect the instance to decide
-			return session.Interceptor.IsTransient(entity) ??
-			       session.GetEntityPersister(entityName, entity).IsTransient(entity, session);
+
+			if (session.Interceptor.IsTransient(entity) == true)
+				return true;
+
+			if (entity is INHibernateProxy proxy && proxy.HibernateLazyInitializer.IsUninitialized)
```

Comment:
Can proxy be at all transient? Can't this check be simplified to `enitity is INHibernateProxy`?